pkg/minishift/hostfolder/hostfolder.go

```diff
@@ -0,0 +1,428 @@
+		return errors.New("No host folders defined.")
+	}
+
+	hostfolder := getHostfolderByName(name)
+	if hostfolder == nil {
+		return fmt.Errorf("No host folder defined as: %s", name)
+	} else {
+		ensureMountPointExists(driver, hostfolder)
+		mountHostfolder(driver, hostfolder)
+	}
+	return nil
+}
+
+// Performs mounting of host folders
+func MountHostfolders(driver drivers.Driver) error {
+	if !isHostRunning(driver) {
+		return errors.New("Host is in the wrong state.")
+	}
+
+	if !IsHostfoldersDefined(true) {
+		return errors.New("No host folders defined.")
+	}
+
+	println("-- Mounting hostfolders")
+
+	hostfolders := instanceState.Config.HostFolders
+	for i := range hostfolders {
+		// Ignore individual errors or aggregate
+		mountHostfolder(driver, &hostfolders[i])
+	}
+
+	return nil
+}
+
+func Umount(driver drivers.Driver, name string) error {
+	if !isHostRunning(driver) {
+		return errors.New("Host is in the wrong state.")
+	}
+
+	if !IsHostfoldersDefined(true) {
+		return errors.New("No host folders defined")
+	}
+
+	hostfolder := getHostfolderByName(name)
+	if hostfolder == nil {
+		return fmt.Errorf("No host folder defined as: %s", name)
+	} else {
+		umountHostfolder(driver, hostfolder)
+	}
+	return nil
+}
+
+func mountHostfolder(driver drivers.Driver, hostfolder *instanceState.HostFolder) error {
+	if hostfolder == nil {
+		return errors.New("Host folder not defined")
+	}
+
+	switch hostfolder.Type {
+	case "cifs":
+		if err := mountCifsHostfolder(driver, hostfolder); err != nil {
+			return err
+		}
+	default:
+		return errors.New("Unsupported host folder type")
+	}
+
+	return nil
+}
+
+func determineHostIp(driver drivers.Driver) (string, error) {
+	instanceip, err := driver.GetIP()
+	if err != nil {
+		return "", err
+	}
+
+	for _, hostaddr := range util.HostIPs() {
+
+		if util.NetworkContains(hostaddr, instanceip) {
+			hostip, _, _ := net.ParseCIDR(hostaddr)
+			if util.IsIPReachable(driver, hostip.String(), false) {
+				return hostip.String(), nil
+			} else {
+				return "", errors.New("Unreachable")
+			}
+		}
+	}
+
+	return "", errors.New("Unknown error occured")
+}
+
+func mountCifsHostfolder(driver drivers.Driver, hostfolder *instanceState.HostFolder) error {
+	// If "Users" is used as name, determine the IP of host for UNC path on startup
+	if hostfolder.Name == "Users" {
+		hostip, _ := determineHostIp(driver)
+		hostfolder.Options["uncpath"] = fmt.Sprintf("//%s/Users", hostip)
+	}
+
+	print(fmt.Sprintf("   Mounting '%s': '%s' as '%s' ... ",
+		hostfolder.Name,
+		hostfolder.Options["uncpath"],
+		hostfolder.Mountpoint()))
+
+	if isMounted, err := isHostfolderMounted(driver, hostfolder); isMounted {
+		println("Already mounted")
+		return fmt.Errorf("Host folder is already mounted. %s", err)
+	}
+
+	if !isCifsHostReachable(driver, hostfolder.Options["uncpath"]) {
+		print("Unreachable\n")
+		return errors.New("Host folder is unreachable")
+	}
+
+	cmd := fmt.Sprintf(
+		"sudo mount -t cifs %s %s -o username=%s,password=%s",
```

Comment:
Whatever works. I also did not test whether it really breaks. However, given that we have seen similar issues in other parts of the code, I thought I point it out. IMO single quotes would cover most problems cases, except where the password contains a single quote itself.

Not sure how the credentials file helps. This is on the VM right? So you need to get the password from the host to the VM, most likely via a ssh command. That's imo where the problem lies. Provided of course the command gets executed in a shell environment. I guess this is not the case when you call mount directly as you do now!?